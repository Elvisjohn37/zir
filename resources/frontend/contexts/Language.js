import React from 'react';
import languageReducer from './language/languageReducer';
import { getCookie } from 'frontend/utils/helper';

let LanguageContext = React.createContext();

export default function LangProvider({ children, config }) {
	let availableProp = config.available || '';
	let disabledProp = config.disabled || '';
	let [langState, langDispatch] = React.useReducer(languageReducer, {
		active: config.force || getCookie(config.key) || config.default,
		available: availableProp.split(','),
		disabled: disabledProp.split(','),
	});

	return (
		<LanguageContext.Provider
			value={{
				langState,
				langDispatch,
			}}
		>
			{children}
		</LanguageContext.Provider>
	);
}

export let LangConsumer = LanguageContext.Consumer;
