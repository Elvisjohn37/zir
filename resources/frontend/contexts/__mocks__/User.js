import React from 'react';
import { componentModuleMock, transformProps } from 'frontend/tests/helpers';
let mockedModules = componentModuleMock(jest.createMockFromModule('../User'));

mockedModules.default = ({ children, ...props }) => {
	let { newProps, childProps } = transformProps(props);

	return (
		<div {...newProps} data-testid="User">
			{children}
			{childProps.map(() => childProps)}
		</div>
	);
};

let userState;
mockedModules.userDispatch = () => {};

// main component
mockedModules.UserConsumer = ({ children }) => {
	return children({ userState, userDispatch: mockedModules.userDispatch });
};

// setter
mockedModules.setUserState = (state) => {
	Object.assign(userState, state);
};

// module resetter
mockedModules.resetUserState = () => {
	userState = {
		isLoading: true,
		isLogin: false,
		isError: false,
		user: {
			parent: 'testParent',
		},
	};
};
mockedModules.resetUserState();

// getter
mockedModules.getUserState = () => {
	return userState;
};

module.exports = mockedModules;
