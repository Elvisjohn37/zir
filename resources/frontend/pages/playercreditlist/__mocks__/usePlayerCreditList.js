import React from 'react';
import {
	getPlayerCreditList,
	setPlayerCreditListData,
	resetPlayerCreditListData,
	getPlayerCreditListData,
} from 'frontend/ajax/__mocks__/backend/getPlayerCreditList';

let mockedModule = jest.createMockFromModule('../usePlayerCreditList.js');

let usePlayerCreditListData;
let isProceedHook = true;

mockedModule.usePlayerCreditList = jest.fn(({ isFetching, setIsFetching, date, page, setCreditListError }) => {
	let [playerCreditList, setPlayerCreditList] = React.useState({
		content: [],
		totalRow: [],
		rowsPerPage: 0,
	});

	if (!isProceedHook) {
		return playerCreditList;
	}

	React.useEffect(() => {
		if (isFetching) {
			getPlayerCreditList({
				date,
				page,
				success: (response) => {
					usePlayerCreditListData = response.data.data;
					setIsFetching(false);
					setPlayerCreditList(response.data.data);
				},
				error: () => {
					setCreditListError(true);
					setIsFetching(false);
				},
			});
		}
	}, [isFetching]);
	return playerCreditList;
});

function setIsProceedHook(isProceed) {
	isProceedHook = isProceed;
}

function setUsePlayerCreditListData(data) {
	usePlayerCreditListData = { ...mockedModule.usePlayerCreditList, ...data };
}

function resetUsePlayerCreditListData() {
	usePlayerCreditListData = {};
	resetPlayerCreditListData();
	isProceedHook = true;
}

function setUsePlayerCreditListResponse(result, data) {
	setPlayerCreditListData(result, data);
}

export default mockedModule.usePlayerCreditList;

export {
	setUsePlayerCreditListData,
	resetUsePlayerCreditListData,
	usePlayerCreditListData,
	setIsProceedHook,
	setUsePlayerCreditListResponse,
	getPlayerCreditListData,
};
