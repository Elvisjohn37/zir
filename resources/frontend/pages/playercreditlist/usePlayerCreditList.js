import React from 'react';
import { getPlayerCreditList } from 'frontend/ajax/backend';

export default function usePlayerCreditList({ 
	isFetching, 
    setIsFetching, 
    date, 
    page, 
    setCreditListError 
}) {
    let [playerCreditList, setPlayerCreditList] = React.useState({
		content: [],
		totalRow: [],
		rowsPerPage: 0,
	});

    React.useEffect(() => {
        if (isFetching) {
			getPlayerCreditList({
				date,
				page,
				success: (response) => {
                    setIsFetching(false);
					setPlayerCreditList(response.data.data);
				},
				error: () => {
					setCreditListError(true);
                    setIsFetching(false);
				},
			});
		}
    }, [isFetching]);

    return playerCreditList;
}
