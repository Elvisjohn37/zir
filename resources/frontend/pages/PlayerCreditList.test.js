import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import PlayerCreditList from './PlayerCreditList';
import * as MediaQuery from 'react-responsive';
import * as usePlayerCreditList from './playercreditlist/usePlayerCreditList';
import { setUserState } from 'frontend/contexts/User';
import userEvent from '@testing-library/user-event';

jest.mock('./playercreditlist/usePlayerCreditList');
jest.mock('@mui/material/Table');
jest.mock('@mui/material/TableHead');
jest.mock('@mui/material/TableRow');
jest.mock('@mui/material/TableBody');
jest.mock('@mui/material/TableCell');
jest.mock('@mui/material/TableContainer');
jest.mock('frontend/components/TableIndicator');
jest.mock('frontend/components/Sidebar');
jest.mock('frontend/components/Language');
jest.mock('frontend/components/ReportCommonFormatter');
jest.mock('frontend/components/ReportCreditFormatter');
jest.mock('frontend/components/TableRefresh');
jest.mock('frontend/contexts/Layout');
jest.mock('frontend/contexts/Language');
jest.mock('frontend/contexts/User');
jest.mock('react-responsive');
jest.mock('react-intl');

describe('<PlayerCreditList />', () => {
	beforeEach(() => {
		jest.spyOn(MediaQuery, 'setMediaQuery');
	});

	afterEach(() => {
		resetMocks();
		cleanup();
	});

	function resetMocks() {
		jest.clearAllMocks();
		jest.restoreAllMocks();
		usePlayerCreditList.resetUsePlayerCreditListData();
		MediaQuery.setMediaQuery(true);
	}

	function getComponentProps(transactionType = 'betting') {
		let params = {
			type: transactionType,
			date: '2021-07-15',
		};
		if (transactionType == 'betting') {
			params = { ...params, productID: 1 };
		}
		return params;
	}

	function expectCorrectTableContent() {
		setUserState({ user: { parent: 'test' } });
		let responseData = usePlayerCreditList.getPlayerCreditListData().data.data;
		let columnsLength = 6;
		responseData.content.forEach((data, index) => {
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 1]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 2]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 3]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 4]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 5]).toBeInTheDocument();
		});
	}

	it('should match snapshot', () => {
		let { asFragment, rerender } = render(<PlayerCreditList match={{ params: getComponentProps() }} />);
		expect(asFragment()).toMatchSnapshot();

		rerender(<PlayerCreditList match={{ params: getComponentProps('promotion') }} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should have table loading indicator on table during data fetching', () => {
		usePlayerCreditList.setIsProceedHook(false);
		render(<PlayerCreditList match={{ params: getComponentProps() }} />);
		expect(screen.getByText('Loading TableIndicator')).toBeInTheDocument();
	});

	it('should display correct data in table after request from backend', () => {
		let { rerender } = render(<PlayerCreditList match={{ params: getComponentProps() }} />);
		expectCorrectTableContent();
		usePlayerCreditList.setUsePlayerCreditListResponse(true, {
			content: [
				{
					dateTime: '2021-07-15 19:32:23.753000',
					creditLimit: '1000.000000',
					playerTotalBalance: '1000.000000',
					playableBalance: '1000.000000',
				},
				{
					dateTime: '2021-07-15 19:32:24.753000',
					creditLimit: '1000.000000',
					playerTotalBalance: '1000.000000',
					playableBalance: '1000.000000',
				},
			],
			totalRow: 2,
			rowsPerPage: 20,
		});
		rerender(<PlayerCreditList match={{ params: getComponentProps() }} />);
		expectCorrectTableContent();
	});

	it('should fetch data from backend when user click the refresh button', () => {
		render(<PlayerCreditList match={{ params: getComponentProps() }} />);
		userEvent.click(screen.getByTestId('MobileTableRefresh'));
		expect(usePlayerCreditList.default).toHaveBeenNthCalledWith(1, {
			isFetching: true,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setCreditListError: expect.any(Function),
		});
		expect(usePlayerCreditList.default).toHaveBeenNthCalledWith(2, {
			isFetching: false,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setCreditListError: expect.any(Function),
		});
		expect(usePlayerCreditList.default).toHaveBeenNthCalledWith(3, {
			isFetching: true,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setCreditListError: expect.any(Function),
		});
		expect(usePlayerCreditList.default).toHaveBeenNthCalledWith(4, {
			isFetching: false,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setCreditListError: expect.any(Function),
		});
		expect(usePlayerCreditList.default).toHaveBeenCalledTimes(4);
	});

	it('should display refresh button base on the type of device', () => {
		let { rerender } = render(<PlayerCreditList match={{ params: getComponentProps() }} />);
		let MobileTableRefresh = screen.getByTestId('MobileTableRefresh');
		expect(MobileTableRefresh).toBeInTheDocument();

		MediaQuery.setMediaQuery(false);
		rerender(<PlayerCreditList match={{ params: getComponentProps() }} />);
		let DesktopTableRefresh = screen.getByTestId('DesktopTableRefresh');
		expect(DesktopTableRefresh).toBeInTheDocument();
	});

	it('should display pagination and record base on selected page', () => {
		render(<PlayerCreditList match={{ params: getComponentProps() }} />);
		expect(screen.getByText('Pagination page: 0')).toBeInTheDocument();
		expect(
			screen.getByText('Pagination count: ' + usePlayerCreditList.usePlayerCreditListData.totalRow)
		).toBeInTheDocument();
		expect(
			screen.getByText('Pagination rowsPerPage: ' + usePlayerCreditList.usePlayerCreditListData.rowsPerPage)
		).toBeInTheDocument();
	});

	it('should display no record found when record is empty', () => {
		usePlayerCreditList.setUsePlayerCreditListResponse(true, { content: [] });
		let { container } = render(<PlayerCreditList match={{ params: getComponentProps() }} />);
		let tableRow = container.getElementsByClassName('testTableRow');
		expect(tableRow).toHaveLength(0);
	});

	it('should display error message when backend response is not valid', () => {
		usePlayerCreditList.setUsePlayerCreditListResponse(false);
		render(<PlayerCreditList match={{ params: getComponentProps() }} />);
		expect(screen.getByText('Error Response')).toBeInTheDocument();
	});
});
