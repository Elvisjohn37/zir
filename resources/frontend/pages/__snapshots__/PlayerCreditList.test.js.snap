// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerCreditList /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="reports"
    data-testid="LangDataProvider"
  >
    <div>
      reports
    </div>
    <div
      data-testid="default"
      sidebargroup="report"
    >
      <div
        class="tableSection"
      >
        <div
          data-testid="MobileTableRefresh"
          isdisabled="false"
        />
        <div>
          <div
            data-testid="TableIndicator"
            id="testTableInit"
            isempty="false"
            iserror="false"
            isloading="false"
            pagination="{\\"count\\":6,\\"page\\":0,\\"rowsPerPage\\":20}"
          >
            <span>
              Pagination count: 6
            </span>
            <span>
              Pagination page: 0
            </span>
            <span>
              Pagination rowsPerPage: 20
            </span>
            <span>
              Done loading TableIndicator
            </span>
            <div
              data-testid="TableContainer"
            >
              <div
                aria-label="a dense table"
                classes="{\\"root\\":\\"table\\"}"
                data-testid="Table"
              >
                <div
                  data-testid="TableHead"
                >
                  <div
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="colHeaderRowNum"
                      data-testid="TableCell"
                    >
                      #
                    </div>
                    <div
                      align="center"
                      class="colHeaderDateTime"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="date"
                      >
                        <div>
                          date
                        </div>
                      </div>
                       & 
                      <div
                        data-testid="FormattedMessage"
                        id="time"
                      >
                        <div>
                          time
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderDescription"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="details"
                      >
                        <div>
                          details
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colCreditLimit"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="creditLimit"
                      >
                        <div>
                          creditLimit
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="colHeaderPlayerTotalBalance"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="playerTotalBalance"
                      >
                        <div>
                          playerTotalBalance
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="colHeaderPlayableBalance"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="playableBalance"
                      >
                        <div>
                          playableBalance
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  data-testid="TableBody"
                >
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      1
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      2
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      3
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      4
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      5
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      6
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<PlayerCreditList /> should match snapshot 2`] = `
<DocumentFragment>
  <div
    category="reports"
    data-testid="LangDataProvider"
  >
    <div>
      reports
    </div>
    <div
      data-testid="default"
      sidebargroup="report"
    >
      <div
        class="tableSection"
      >
        <div
          data-testid="MobileTableRefresh"
          isdisabled="false"
        />
        <div>
          <div
            data-testid="TableIndicator"
            id="testTableInit"
            isempty="false"
            iserror="false"
            isloading="false"
            pagination="{\\"count\\":6,\\"page\\":0,\\"rowsPerPage\\":20}"
          >
            <span>
              Pagination count: 6
            </span>
            <span>
              Pagination page: 0
            </span>
            <span>
              Pagination rowsPerPage: 20
            </span>
            <span>
              Done loading TableIndicator
            </span>
            <div
              data-testid="TableContainer"
            >
              <div
                aria-label="a dense table"
                classes="{\\"root\\":\\"table\\"}"
                data-testid="Table"
              >
                <div
                  data-testid="TableHead"
                >
                  <div
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="colHeaderRowNum"
                      data-testid="TableCell"
                    >
                      #
                    </div>
                    <div
                      align="center"
                      class="colHeaderDateTime"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="date"
                      >
                        <div>
                          date
                        </div>
                      </div>
                       & 
                      <div
                        data-testid="FormattedMessage"
                        id="time"
                      >
                        <div>
                          time
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderDescription"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="details"
                      >
                        <div>
                          details
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colCreditLimit"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="creditLimit"
                      >
                        <div>
                          creditLimit
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="colHeaderPlayerTotalBalance"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="playerTotalBalance"
                      >
                        <div>
                          playerTotalBalance
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="colHeaderPlayableBalance"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="playableBalance"
                      >
                        <div>
                          playableBalance
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  data-testid="TableBody"
                >
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      1
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      2
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      3
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      4
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      5
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      6
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                      />
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testCreditLimit"
                        data-testid="CreditLimit"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayerTotalBalance"
                        data-testid="PlayerTotalBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
